### DISCUSSION TITLE
Tripadvisor integration for photos, reviews and ratings

### DISCUSSION THREAD

**Discussion by Ihor-Khomenko** [organicmaps/organicmaps#772] [Ideas]
Hi! None of OSM maps offer proper and up to date photos and reviews for places on the map such as parks, restaurants, businesses, etc. 

Tripadvisor offers an API for developers and anyone can use it for their app or website - https://developer-tripadvisor.com/content-api/

There are might be privacy concerns, so additional steps to anonymize user inside OsmAnd might be necessary. 

I have attached a video demonstration on how it works on Apple Maps. 

https://user-images.githubusercontent.com/39989281/123804461-e970c100-d8f5-11eb-967a-470304ac39ed.mp4

**Comment by biodranik** [comment]
What is your use-case for this feature? Why you can't use other apps for that?

**Reply by Ihor-Khomenko** [reply]
Because it’s much more convenient to see photos and people’s feedback about a certain spot in single app and not switching back and forth between these apps. For example, i can use maps app to see some interesting spots and add it to favourites. 
That will give you a massive killer feature over mapsme and osmand. As for me, i want completely switch from google maps to osm maps app. And now it’s difficult mainly because Organic and other apps lack photos and feedback.

**Reply by georg-d** [reply]
Switching between apps is tedious. It means the POI / location _is already open_ in one app, but still, _you need to find the POI / location_ in the second app - and of course, sharing the location does often enough not work because the intends, coordinate formats etc are not compatible with each other (e.g. because of German android, an app shares N 5, O 5 but the other expects N 5, E 5, or think about x.x° vs x° x.x' vs x° x' x.x").

**Reply by VladimirMorozov** [reply]
If one of the problems is finding the POI in another app, then maybe OrganicMaps could incorporate a link instead of dragging the whole functionality with pictures and comments inside itself. Maybe load the overall rating, which could be an online feature, not more. Then you can continue browsing restaurants in TripAdvisor.

**Comment by amo-f77** [comment]
I would not want this feature in an **offline** map app. Please keep it simple and not bloated. 
If I wanted photos and reviews I'd use TA or google maps.

**Reply by Ihor-Khomenko** [reply]
I agree, that offline map shouldn’t be overloaded. On the other side, when you planning your trip, you usually use several apps and services to plan your routes and save places to visit. And when you planning it, usually, you have an internet connection. 
Why using google maps, tripadvisor, wikiloc etc, when you can use only organic maps with photos and reviews feature and wiki loc to import some trails to the app?

**Reply by georg-d** [reply]
I see the point of both. To pick one of the two, IMHO two general strategic decisions are required: 

1. Shall OrganicMaps only contain 100% offline features except downloading the data (maps) needed for offline use? IMHO not, because that leaved most of a smartphone's potential unused, e.g. we'd need to remove  the possibility to update/add and upload OSM changes. And we'd never get e.g. traffic data - I am still looking for an OSM-based, offline-capable, and also general-purpose-capable map app with traffic info taken into account while navigating and rendering the map. Having online features in an offline-capable app is no contradiction as the core of OM works offline, only some features need an internet connection, and they might cache the received data so we can still use them when being offline some moments later - just with data that is 5 minutes old.
2. Shall OM only incorporate features that are really simple to use or also get more advanced features? In different words, shall OgranicMaps become as powerful but also as complex as OsmAnd? Personally, I'm fine with both as I use both. Advanced features could be added but require users to un-hide / activate them in the configuration, so users looking for an app that is simple to use won't be disturbed by them. But even when initially hidden, they make the app complex to implement, maintain, learn (that the advanced feature is there), configure, support.

**Comment by georg-d** [comment]
Did you try OpenPlaceReviews in OsmAnd? I did not yet but from what I read, I have the impression it is - to a considerable degree - the feature set you look for.

**Reply by Ihor-Khomenko** [reply]
> 
> 
> Did you try OpenPlaceReviews in OsmAnd? I did not yet but from what I read, I have the impression it is - to a considerable degree - the feature set you look for.

Hi, i didn't now that OSMand has this feature. I stopped using OSMand (i have premium) because of Organic Maps. But i will check it out. Thank you.

**Reply by kudlav** [reply]
OpenPlaceReviews is OSMand project and it is still a draft. So the future of this project is unsure.

**Reply by georg-d** [reply]
> the future of this project is unsure.

Just like the future of any project ;-) Even mature, well usable projects that are used intensely and owned by well fundend organizations may be stopped, see e.g. https://en.wikipedia.org/wiki/Panoramio 

The more it is used, the more it becomes shaped and it's future "ensured".

**Reply by kudlav** [reply]
Anyway, it's not a _feature set, you can use_. That's the point of my post. I was searching more info about this project and the most recent posts I've found were quite sceptical.

**Comment by j5k** [comment]
I would prefer the integration of wikidata instead of using a closed source service. Like OSMand, but maybe with other informations too.

**Comment by FinixFighter** [comment]
This seems a duplicate of #371

**Comment by TheAdventurer64** [comment]
I like this idea, but wouldn't this go against privacy guidelines?

Edit: I found [this comment](https://github.com/organicmaps/organicmaps/discussions/1725#discussioncomment-1860475). from one of the developers. If storing data from Booking.com wouldn't be possible, how would TripAdvisor be any different?

**Comment by TheAdventurer64** [comment]
See [this comment](https://github.com/organicmaps/organicmaps/discussions/1725#discussioncomment-1860475).

**Reply by FinixFighter** [reply]
In fact I think that the best (and maybe only) possible solution for adding ratings/reviews is using open alternatives, like http://openplacereviews.org/ for instance.

**Reply by TheAdventurer64** [reply]
> In fact I think that the best (and maybe only) possible solution for adding ratings/reviews is using open alternatives, like http://openplacereviews.org/ for instance.

Are these platforms big enough to be supported by Organic Maps?

**Reply by FinixFighter** [reply]
> Are these platforms big enough to be supported by Organic Maps?

What do you mean? Well, they're not big tech of course, they're still in development, but it looks very promising imo, and a collaboration would be beneficial for both. 
Anyway some features are already available, like uploading pictures for places (outdoor, indoor, menu) for instance, which is already implemented in OsmAnd. That could be a good starting point.

**Reply by kudlav** [reply]
The project openplacereviews is doubtful. There are no commits and no reaction from developers (how to help this project, use this project, to the support questions, ...)

See this post, still without any reaction
https://t.me/openplacereviews/1047
![obrazek](https://user-images.githubusercontent.com/11516259/160232652-6e6636fe-0427-49f4-afc7-4b67d1ba2a89.png)

**Reply by FinixFighter** [reply]
> The project openplacereviews is doubtful. There are no commits and no reaction from developers (how to help this project, use this project, to the support questions, ...)
> 
> See this post, still without any reaction https://t.me/openplacereviews/1047 ![obrazek](https://user-images.githubusercontent.com/11516259/160232652-6e6636fe-0427-49f4-afc7-4b67d1ba2a89.png)

You're right. I hope the project will go on (the original one or eventually a fork of it).

**Reply by TheAdventurer64** [reply]
Maybe as a placeholder for now, the ability to vote a place up or down, similar to Apple Maps:![image](https://user-images.githubusercontent.com/50059322/160276679-70a1bd3d-ee82-4081-996f-83ea565e23bd.jpeg)

**Reply by biodranik** [reply]
Any simplified integral rating would not reflect reality. It would be more useful to avoid using ratings and only allow detailed comments with votes for comments.

In any case, that's a completely different subproject that should be implemented carefully.

**Comment by contrapunctus-1** [comment]
https://mangrove.reviews/ should be used for unrestricted ("free") and possibly-offline reviews and ratings.

https://wiki.openstreetmap.org/wiki/Wikimedia_Commons should be used for unrestricted ("free") and possibly-offline photos.
